tests/System.Text.Primitives.Tests/ParserTests.cs

```diff
@@ -4,472 +4,983 @@
 
 namespace System.Text.Primitives.Tests
 {
-    public class ParserTests
+	public class ParserTests
     {
-        private byte[] UtfEncode(string s)
+		private byte[] UtfEncode(string s)
         {
             return Encoding.UTF8.GetBytes(s);
         }
-        
-        #region ulong
-        [Theory]
-        [InlineData("1728", true, 0, 1728, 4)]
-        [InlineData("blahblahh1751110", true, 9, 1751110, 7)]
-        [InlineData("987abcdefg", true, 0, 987, 3)]
-        [InlineData("The biggest ulong is 9223372036854775808.", true, 21, 9223372036854775808, 19)]
+        private byte[] UtfEncode(string s, bool utf16)
+        {
+            if (utf16)
+                return Encoding.Unicode.GetBytes(s);
+            else
+                return UtfEncode(s);
+        }
+
+		#region byte
+
+		[Theory]
+        [InlineData("55", true, 0, 55, 2)]
```

Comment:
I see that the last parameter will differ for Utf8/Utf16 cases (by a factor of two), but I wonder if you couldn't somehow still re-use the test data for both cases.

In the end it doesn't matter a whole lot because this is a generated file, but it would help reading it a bit :smile: .